core/src/main/java/org/infinispan/interceptors/impl/NotificationInterceptor.java

```diff
@@ -18,6 +18,14 @@
  */
 public class NotificationInterceptor extends DDSequentialInterceptor {
    private CacheNotifier notifier;
+   private ReturnHandler transactionCompleteReturnHandler = (ctx1, command1, rv, throwable) -> {
+      if (throwable != null)
+         return null;
```

Comment:
Yep, they are exactly the same, when `throwable != null`.

I've wanted to add specialized interfaces for success/exception/finally, but things get complicated very soon if you want to avoid creating 2 lambdas for a single interceptor, so I wanted to get a better idea of how these are used. Also, I'm kind of hoping to merge `ReturnHandler` and `ForkReturnHandler` at the same time, to avoid having 8 interfaces.